homeassistant/components/recorder/purge.py

```diff
@@ -33,10 +33,16 @@ def purge_old_data(instance, purge_days, repack):
             )
             _LOGGER.debug("Deleted %s events", deleted_rows)
 
-        # Execute sqlite vacuum command to free up space on disk
-        if repack and instance.engine.driver in ("pysqlite", "postgresql"):
-            _LOGGER.debug("Vacuuming SQL DB to free space")
-            instance.engine.execute("VACUUM")
+        if repack:
+            # Execute sqlite or postgresql vacuum command to free up space on disk
+            if instance.engine.driver in ("pysqlite", "postgresql"):
+                _LOGGER.debug("Vacuuming SQL DB to free space")
+                instance.engine.execute("VACUUM")
+            # Optimize mysql / mariadb tables to free up space on disk
+            elif instance.engine.driver in ("mysqldb"):
+                _LOGGER.debug("Optimizing SQL DB to free space")
+                instance.engine.execute("OPTIMIZE TABLE states")
+                instance.engine.execute("OPTIMIZE TABLE events")
```

Comment:
Optimize table can optimize multiple tables at once.
```suggestion
                instance.engine.execute("OPTIMIZE TABLE states, events")
```